mpu6050/mpu6050.c

```diff
@@ -4,60 +4,106 @@
 #include <linux/err.h>
 #include <linux/i2c.h>
 #include <linux/i2c-dev.h>
+#include <linux/timekeeping.h>
+#include <linux/miscdevice.h>
 
 #include "mpu6050-regs.h"
 
 
+enum {
+	X = 0,
+	Y = 1,
+	Z = 2
+};
+
+#define SAMPLES_COUNT (10)
+
 struct mpu6050_data {
-	struct i2c_client *drv_client;
 	int accel_values[3];
 	int gyro_values[3];
 	int temperature;
+	unsigned long timestamp;
 };
 
-static struct mpu6050_data g_mpu6050_data;
+static struct i2c_client *mpu6050_drv_client;
+struct mpu6050_data mpu6050_data_samples[SAMPLES_COUNT];
+int mpu6050_current_sample_idx;
```

Comment:
Maybe declare these variables static?

Reply:
Sure, this has to be static. Nicely spotted